_monitoring-your-cluster/logs.md

```diff
@@ -0,0 +1,205 @@
+[2019-10-18T16:52:51,297][DEBUG][o.o.i.r.TransportReindexAction] [node1] [1626]: preparing bulk request
+[2019-10-18T16:52:51,299][DEBUG][o.o.i.r.TransportReindexAction] [node1] [1626]: sending [3] entry, [222b] bulk request
+[2019-10-18T16:52:51,310][INFO ][o.e.c.m.MetaDataMappingService] [node1] [some-new-index/R-j3adc6QTmEAEb-eAie9g] create_mapping [_doc]
+[2019-10-18T16:52:51,383][DEBUG][o.o.i.r.TransportReindexAction] [node1] [1626]: got scroll response with [0] hits
+[2019-10-18T16:52:51,384][DEBUG][o.o.i.r.WorkerBulkByScrollTaskState] [node1] [1626]: preparing bulk request for [0s]
+[2019-10-18T16:52:51,385][DEBUG][o.o.i.r.TransportReindexAction] [node1] [1626]: preparing bulk request
+[2019-10-18T16:52:51,386][DEBUG][o.o.i.r.TransportReindexAction] [node1] [1626]: finishing without any catastrophic failures
+[2019-10-18T16:52:51,395][DEBUG][o.o.i.r.TransportReindexAction] [node1] Freed [1] contexts
+```
+
+The DEBUG and TRACE levels are extremely verbose. If you enable either one to troubleshoot a problem, disable it after you finish.
+
+There are other ways to change log levels:
+
+1. Add lines to `opensearch.yml`:
+
+   ```yml
+   logger.org.opensearch.index.reindex: debug
+   ```
+
+   Modifying `opensearch.yml` makes the most sense if you want to reuse your logging configuration across multiple clusters or debug startup issues with a single node.
+
+2. Modify `log4j2.properties`:
+
+   ```
+   # Define a new logger with unique ID of reindex
+   logger.reindex.name = org.opensearch.index.reindex
+   # Set the log level for that ID
+   logger.reindex.level = debug
+   ```
+
+   This approach is extremely flexible, but requires familiarity with the [Log4j 2 property file syntax](https://logging.apache.org/log4j/2.x/manual/configuration.html#Properties). In general, the other options offer a simpler configuration experience.
+
+   If you examine the default `log4j2.properties` file in the configuration directory, you can see a few OpenSearch-specific variables:
+
+   ```
+   appender.console.layout.pattern = [%d{ISO8601}][%-5p][%-25c{1.}] [%node_name]%marker %m%n
+   appender.rolling_old.fileName = ${sys:os.logs.base_path}${sys:file.separator}${sys:os.logs.cluster_name}.log
+   ```
+
+   - `${sys:os.logs.base_path}` is the directory for logs (for example, `/var/log/opensearch/`).
+   - `${sys:os.logs.cluster_name}` is the name of the cluster.
+   - `[%node_name]` is the name of the node.
+
+
+## Slow logs
+
+OpenSearch has two *slow logs*, logs that help you identify performance issues: the search slow log and the indexing slow log.
+
+These logs rely on thresholds to define what qualifies as a "slow" search or indexing operation. For example, you might decide that a query is slow if it takes more than 15 seconds to complete. Unlike application logs, which you configure for modules, you configure slow logs for indexes. By default, both logs are disabled (all thresholds are set to `-1`):
```

Comment:
Just to make it perfectly clear that a threshold is applied to the two types of logs mentioned in the previous sentence.

These logs rely on thresholds to define what qualifies as a "slow" search or a "slow" indexing operation.